spring-cloud-azure-appconfiguration-config/src/main/java/com/microsoft/azure/spring/cloud/config/AzureCloudConfigWatch.java

```diff
@@ -98,16 +112,44 @@ private void refresh(ConfigStore store, String storeSuffix, String watchedKeyNam
         if (firstTimeMap.get(storeName) == null) {
             storeEtagMap.put(storeName, etag);
             firstTimeMap.put(storeName, false);
-            return;
+            propertyCache.updateRefreshCacheTime(store.getName(), watchedKeyNames, delay);
         }
+        
+        if (!etag.equals(storeEtagMap.get(storeName))) {
+            Date date = new Date();
+
+            // Checks all cached items to see if they have been updated
+            List<String> refreshKeys = propertyCache.getRefreshKeys(store.getName());
+            for (int i = 0; i < refreshKeys.size(); i++) {
+                String refreshKey = refreshKeys.get(i);
+                if (refreshKey.contains(watchedKeyNames.replace("*", ""))) {
```

Comment:
Just question, the contains compare is different with regexp match, though regexp much requires some performance cost, will this contain match cases not expected? If yes, or we tolerate such match?